knimin/lib/data_access.py

```diff
@@ -3532,6 +3533,204 @@ def read_dna_plate(self, dna_plate_id):
             # Magic number 0 -> there is only 1 result row
             return dict(res[0])
 
+    def normalize_shotgun_plate(self, shotgun_plate_id, email, echo,
+                                plate_normalization_sample,
+                                plate_normalization_water):
+	"""Adds a normalized shotgun plate to the system
+
+	Parameters
+	----------
+	shotgun_plate_id : int
+	    The shotgun plate id
+	email : str
+	    The email of the user
+	echo : str
+	    The echo machine performing the normalization
+	plate_normalization_sample : 2D np.array of float
+	    The sample volume in nanoliters for each well
+	plate_normalization_water : 2D np.array of float
+	    The water volume in nanoliters for each well
+
+	Raises
+	------
+	ValueError
+	    If the shotgun plate `shotgun_plate_id` does not exist
+	    If `plate_normalization_sample` dimensions doesn't match the plate
+	        type
+	    If `plate_normalization_water` dimensions doesn't match the plate
+	        type
+	    If the echo instrument does not exist
+
+	Returns
+	-------
+	int
+	    The created normalized_shotgun_plate_id
+	"""
+        with TRN:
+            # verify the plate exists
+            sql = """SELECT EXISTS(
+                        SELECT 1
+                        FROM pm.shotgun_plate
+                        WHERE shotgun_plate_id = %s)"""
+            TRN.add(sql, [shotgun_plate_id])
+            res = TRN.execute_fetchindex()[0]
```

Comment:
Minor, but you can use `TRN.execute_fetchlast()`. is equivalent to `TRN.execute_fetchindex()[0][0]`